packages/flutter_tools/bin/macos_assemble.sh

```diff
@@ -93,3 +93,12 @@ RunCommand "${FLUTTER_ROOT}/bin/flutter"                                    \
     --build-outputs="${build_outputs_path}"                                 \
     --output="${ephemeral_dir}"                                             \
    "${build_mode}_macos_bundle_flutter_assets"
+
+if [[ $? -ne 0 ]]; then
```

Comment:
I forget, is there something blocking us from merging this into the Dart script we're using for Windows and Linux? It would be nice to go in that direction instead of adding even more logic to the shell script version.

Reply:
No, its do-able with some light refactoring besides the tripwire. If we can resolve that its 5 mins of work

Reply:
Couldn't we combine scripts, and have the tripwire logic based on OS (if we really need the tripwire here rather than in the project, which per my other comment I'm still not convinced of)

Reply:
> Couldn't we combine scripts, and have the tripwire logic based on OS

We could for new projects, with some backwards compat for macos_assemble (unless we add the project migration support, but like I said I don't really have time for that this week)

Reply:
I was just thinking converting `macos_assemble.sh` into a thin `tool_backend.dart` wrapper, not doing an actual migration at this point. Just so we're not adding more bash logic and continuing to maintain bash complexity, when we already have a Dart version that's more testable and more maintainable basically ready to swap in.